openstack/compute/v2/servers/requests.go

```diff
@@ -95,6 +96,29 @@ type Network struct {
 	FixedIP string
 }
 
+// Personality is an array of files that are injected into the server at launch.
+type Personality []*File
+
+// File is used within CreateOpts and RebuildOpts to inject a file into the server at launch.
+type File struct {
+	// Path of the file
+	Path string
+	// Contents of the file. Maximum content size is 255 bytes.
+	Contents []byte
+}
+
+// MarshalJSON marshals the escaped file, base64 encoding the contents.
+func (f *File) MarshalJSON() ([]byte, error) {
```

Comment:
The file base64 encode process is used by `ToServerCreateMap` and `ToServerRebuildMap`. Having `File` implement `MarshalJSON` seemed like a good way to reduce putting the encode logic in two places and have a `File` know how to marshal itself.